engine/engine-script-library.py

```diff
@@ -33,104 +33,124 @@ def abort_error(msg, sync):
+    waiting for a particular condition, and aborting the roadblock.
+
+    :param label: A label for the roadblock
+    :param timeout: The timeout value for the roadblock
+    :param messages: (Optional) A path to a file containing user messages to send
+    :param wait_for: (Optional) A command to wait for before proceeding with the roadblock
+    :param abort: (Optional) If True, sends an abort message to the roadblock
+    :return: The exit code of the roadblock command
+    """
+
+    # Set the leader for the roadblock to be the "controller"
     leader = "controller"
-    message = ""
-    wait_for = ""
-    wait_for_log = ""
-    do_abort = 0
-
-    arg_iter = iter(args)
-    for arg in arg_iter:
-        if arg == "messages":
-            message = next(arg_iter)
-            print(f"Going to send this user message file: {message}")
-        elif arg == "wait-for":
-            wait_for = next(arg_iter)
-            wait_for_log = tempfile.mkstemp()[1]
-            print(f"Going to run this wait-for command: {wait_for}")
-            print(f"Going to log wait-for to this file: {wait_for_log}")
-        elif arg == "abort":
-            do_abort = 1
-            print("Going to send an abort")
-        else:
-            exit_error(f"[ERROR]do_roadblock() encountered invalid optional parameter(s) {args}")
-
-    if not message:
-        print("Not going to send a user message file")
-    if not wait_for:
-        print("Not going to use wait-for")
-    if not label:
-        exit_error("[ERROR]do_roadblock() label not provided")
-    if not timeout:
-        exit_error("[ERROR]do_roadblock() timeout not provided")
-
-    msgs_log_file = f"{roadblock_msgs_dir}/{label}.json"
-    cmd = []
+    
+    wait_for_log = None
+
+    # If messages is not None, print a message indicating that the user message file is going to be sent
+    if messages is not None:
+        print("Going to send this user message file:", messages)
+
+    # If wait_for is not None, print a message indicating that the wait-for command is going to be run
+    # and set wait_for_log to a temporary file path
+    if wait_for is not None:
+        print("Going to run this wait-for command:", wait_for)
+        wait_for_log = tempfile.mktemp()
+
+    # Raise a ValueError if the label or timeout are not provided
+    if label is None:
+        raise ValueError("[ERROR]do_roadblock() label not provided")
+
+    if timeout is None:
+        raise ValueError("[ERROR]do_roadblock() timeout not provided")
+
+    # Set the path for the user messages log file
+    msgs_log_file = os.path.join(roadblock_msgs_dir, f"{label}.json")
+
+    # Initialize the command string and set the role to "follower"
+    cmd = ""
     role = "follower"
 
-    # Replace rickshaw_host with the appropriate hostname or IP address
-    rickshaw_host = "localhost"
+    # Use subprocess to run a "ping" command to the rickshaw host
     subprocess.run(["ping", "-w", "10", "-c", "4", rickshaw_host])
```

Comment:
I've been looking into [Fabric](https://www.fabfile.org/)/[Invoke](https://www.pyinvoke.org/) for some similar stuff, what do you think of them instead of using subprocess directly?  I'm mainly curious here with this question...